Upgrade style

ldgeo opened the pull request:
- upgrade style example to latest version
- upgrade mapbox-gl-js
- move index example to its own file
- add documentation to download fonts and serve them with Postile
- remove legacy URL's referencing `tilehosting` webservices
- Centering style to `Lyon` by default

ldgeo committed:
move html example to a file

ldgeo committed:
remove initial position from js code

because it's already mentioned in the style

ldgeo committed:
remove unused sprite URL

ldgeo committed:
notify the new fonts option from Postile

ldgeo committed:
upgrade style according to openmaptiles master branch

ldgeo committed:
upgrade mabox-gl-js to latest version

autra commented on index.html:
```diff
@@ -0,0 +1,29 @@
+
+<div id='map'></div>
+<script>
+mapboxgl.accessToken = ' ';
```
do we need an access token here? If yes, maybe mention it in the doc.

ldgeo replied:
no we don't need access token, i remove it now. thanks.

autra reviewed (approved):
I didn't test yet, but from what I can see, LGTM :-)

ldgeo committed:
remove unneeded access token parameter

ldgeo committed:
don't import osm data in DIFF mode

i realized that the default value is diff mode, we don't want that.
https://github.com/openmaptiles/import-osm/blob/master/import_osm.sh#L7

ldgeo commented:
Added one more commit about the `-diff` mode used in the `import-osm` container.
